### PR TITLE
[Blazor] Get rid of the custom QueryStringHelper implementation

### PR DESCRIPTION
**Pull request by javiercn**
Gets rid of the custom QueryStringHelper implementation in the authentication package in favor of the QueryStringEnumerable which is shared source between Components and ASP.NET Core.

Added an ifdef to skip nullability checks in one API and filed a separate issue to track annotating the assembly.
https://github.com/dotnet/aspnetcore/issues/42533

### CODE REVIEW

File: src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs [hunk 1]
```diff
@@ -329,14 +330,37 @@ private string GetReturnUrl(TAuthenticationState state, string defaultReturnUrl
...
+        if (queryStartPos >= 0)
+        {
+            var queryEndPos = url.IndexOf('#', queryStartPos);
+            query = url.AsMemory(queryStartPos..(queryEndPos < 0 ? url.Length : queryEndPos));
+        }
+
+        foreach (var parameter in new QueryStringEnumerable(query))
+        {
+            var decodedName = parameter.DecodeName().Span;
+            if (MemoryExtensions.Equals(parameterName, decodedName, StringComparison.OrdinalIgnoreCase))
```
Comment (SteveSandersonMS) [inline src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs:358]: If you think there's no legitimate case for encoding the `returnUrl` parameter **name** in a special way, then you could avoid an allocation by not decoding it before doing this comparison.

However this is not exactly perf-critical code (it doesn't run super-frequently as far as I can guess) so I think it's fine either way.

Comment (SteveSandersonMS) [reply src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs:358]: Oops, no, forget that. If the value doesn't need decoding, it won't cause an allocation anyway as `DecodeName` is smart enough to return the original value directly.

**Review by SteveSandersonMS — APPROVED** [review]

File: src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs [hunk 2]
```diff
@@ -329,14 +330,37 @@ private string GetReturnUrl(TAuthenticationState state, string defaultReturnUrl
...
+            {
+                return new string(parameter.DecodeValue().Span);
```
[the remark this answers is not in the document]
Comment (javiercn) [reply src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs:360]: Not sure I understand, that's returning the value.

File: src/Shared/QueryStringEnumerable.cs [hunk 3]
```diff
@@ -30,7 +30,11 @@ namespace Microsoft.AspNetCore.Internal;
     /// Constructs an instance of <see cref="QueryStringEnumerable"/>.
     /// </summary>
     /// <param name="queryString">The query string.</param>
+#if DISABLE_NULLABLE
```
[the remark this answers is not in the document]
Comment (javiercn) [reply src/Shared/QueryStringEnumerable.cs:33]: I was trying to "fix" my build errors with minimal impact, bth, I still struggle with nullable in c#.

File: src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (BrennanConroy) [reply src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs:360]: `DecodeValue` may allocate: https://source.dot.net/#Microsoft.AspNetCore.WebUtilities/QueryStringEnumerable.cs,97
So why call it twice on the same value when you already have the decoded span in a local variable?

[the remark this answers is not in the document]
Comment (javiercn) [reply src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs:360]: But what I have on the local variable is the decoded name, not the value. Am I just missing something obvious? (Its Friday after all 😄)

[the remark this answers is not in the document]
Comment (javiercn) [reply src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs:360]: @BrennanConroy this is why you don't do CR before coffe 😄

[the remark this answers is not in the document]
Comment (BrennanConroy) [reply src/Components/WebAssembly/WebAssembly.Authentication/src/RemoteAuthenticatorViewCore.cs:360]: Drinking it now ☕

File: src/Shared/QueryStringEnumerable.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (javiercn) [reply src/Shared/QueryStringEnumerable.cs:33]: I'm going to undo this change as part of https://github.com/dotnet/aspnetcore/issues/42533

**Pull request merged**